SshAgentLib/PpkFormatter.cs

```diff
@@ -301,6 +317,94 @@ public override object Deserialize(Stream aStream)
+              $"{keyDerivationKey} expected");
+          }
+
+          algorithm = pair[1].Trim();
+          try {
+            fileData.kdfAlgorithm = (KeyDerivation)Enum.Parse(typeof(KeyDerivation), algorithm);
+          }
+          catch (Exception e) when (e is ArgumentException || e is OverflowException) {
+            throw new PpkFormatterException(PpkFormatterException.PpkErrorType.FileFormat,
+              "unsupported key derivation algorithm");
+          }
+
+
+          if (!algorithm.StartsWith("Argon2")) {
+            throw new NotImplementedException();
+          }
+
+
+          line = reader.ReadLine();
+          pair = line.Split(delimArray, 2);
+          if (pair[0] != argonMemoryKey) {
+            throw new PpkFormatterException(PpkFormatterException.PpkErrorType.FileFormat,
+              argonMemoryKey + "expected");
+          }
+
+          try {
+            fileData.kdfParameters[argonMemoryKey] = int.Parse(pair[1].Trim());
+          }
+          catch (Exception e) when (e is ArgumentNullException || e is FormatException || e is OverflowException) {
+            throw new PpkFormatterException(PpkFormatterException.PpkErrorType.FileFormat,
+              "expected an int");
+          }
+
+
+          line = reader.ReadLine();
+          pair = line.Split(delimArray, 2);
+          if (pair[0] != argonPassesKey) {
+            throw new PpkFormatterException(PpkFormatterException.PpkErrorType.FileFormat,
+              argonPassesKey + "expected");
+          }
+
+          try {
+            fileData.kdfParameters[argonPassesKey] = int.Parse(pair[1].Trim());
+          }
+          catch (Exception e) when (e is ArgumentNullException || e is FormatException || e is OverflowException) {
+            throw new PpkFormatterException(PpkFormatterException.PpkErrorType.FileFormat,
+              "expected an int");
+          }
+
+          line = reader.ReadLine();
+          pair = line.Split(delimArray, 2);
+          if (pair[0] != argonParallelismKey) {
+            throw new PpkFormatterException(PpkFormatterException.PpkErrorType.FileFormat,
+              argonParallelismKey + "expected");
+          }
+
+          try {
+            fileData.kdfParameters[argonParallelismKey] = int.Parse(pair[1].Trim());
+          }
+          catch (Exception e) when (e is ArgumentNullException || e is FormatException || e is OverflowException) {
+            throw new PpkFormatterException(PpkFormatterException.PpkErrorType.FileFormat,
+              "expected an int");
+          }
+
+          line = reader.ReadLine();
+          pair = line.Split(delimArray, 2);
+          if (pair[0] != argonSaltKey) {
+            throw new PpkFormatterException(PpkFormatterException.PpkErrorType.FileFormat,
+              argonSaltKey + "expected");
+          }
+
+          try {
+            fileData.kdfParameters[argonSaltKey] = Util.FromHex(pair[1].Trim());
+          }
+          catch (ArgumentException e) {
```

Comment:
e is unused and can be removed (causing CI to fail since I enabled warnings as errors)

```suggestion
          catch (ArgumentException) {
```